maykin_2fa/views.py

```diff
@@ -46,6 +46,11 @@ def get_redirect_url(self):
 
     def get_context_data(self, form, **kwargs):
         context = super().get_context_data(form, **kwargs)
+
+        # upstream doesn't provide a value for the "next" context variable at all
+        redirect_to = self.request.GET.get(self.redirect_field_name, "")
```

Comment:
- Why isn't the redirect defined in `get_redirect_url`?
- Shouldn't we use `url_has_allowed_host_and_scheme` to validate the redirect?

Reply:
* get_redirect_url takes it from `request.POST` or `request.GET`, but those params are empty because they're not passed to the template during the `GET` phase of the form

* about `url_has_allowed_host_and_scheme` - yes, this happens too when consuming the redirect URL and actually doing the URL! django-two-factor-auth behaves correctly here. I have added explicit tests for this in Open Forms too to be sure :)

Reply:
Good to know about this behavior of `get_redirect_url`.